pkg/apis/core/validation/validation.go

```diff
@@ -1533,6 +1533,34 @@ func validateStorageOSPersistentVolumeSource(storageos *core.StorageOSPersistent
 	return allErrs
 }
 
+// ValidatePVSecretReference check whether provided SecretReference object is valid in terms of secret name and namespace.
+// This also take a flag to validate the secret name against DNSSubDomainformat (<253char) and DNS1123Label (<63char) format
+// as its required by some PV sources today, ex: CSI. This flag can help on racheting validation of secretRef field.
+func ValidatePVSecretReference(secretRef *core.SecretReference, allowDNSSubDomainFormat bool, fldPath *field.Path) field.ErrorList {
+	var allErrs field.ErrorList
+	if len(secretRef.Name) == 0 {
+		allErrs = append(allErrs, field.Required(fldPath.Child("name"), ""))
+	} else {
+		if !allowDNSSubDomainFormat {
+			for _, msg := range validation.IsDNS1123Label(secretRef.Name) {
+				allErrs = append(allErrs, field.Invalid(fldPath, secretRef.Name, msg))
+			}
+		} else {
```

Comment:
This code path is not used in this PR. Do we have an example where this might be used (e.g. an existing validation for another PV source)?

Reply:
dropped it for now.